lambdas/common/deployment_utils.py

```diff
@@ -0,0 +1,121 @@
+#!/usr/bin/env python
+# -*- encoding: utf-8 -*-
+"""
+Shared library to help surface ECS deployment information.
+"""
+
+import collections
+import datetime
+
+from ecs_utils import get_cluster_arns, get_service_arns, describe_service
+
+
+Deployment = collections.namedtuple(
+    'Deployment',
+    'deployment_key deployment_status color created_at task_definition'
+)
+DeploymentKey = collections.namedtuple('DeploymentKey', 'id service_arn')
+
+
+DATE_FORMAT = "%Y-%m-%dT%H:%M:%S.%fZ"
+
+def _create_deployment_tuple_from_item(item):
+    item_date = datetime.datetime.strptime(item['created_at'], DATE_FORMAT)
+
+    return Deployment(
+        DeploymentKey(item['deployment_id'], item['service_arn']),
+        item['deployment_status'],
+        item['color'],
+        item_date,
+        item['task_definition']
+    )
+
+
+def _create_deployment_tuple_from_ecs(service, deployment):
+    """Takes am AWS ECS API Service & Deployment, returns a namedtuple."""
+    deployment_status = deployment['status']
+    ongoing_deployment = len(service['deployments']) > 1
+
+    color = "blue"
+    if ongoing_deployment:
+        color = "green" if(deployment_status == "PRIMARY") else "blue"
+
+    return Deployment(
+        DeploymentKey(deployment['id'], service['serviceArn']),
+        deployment_status,
+        color,
+        deployment['createdAt'],
+        deployment['taskDefinition']
+    )
+
+
+def _get_service_deployments(ecs_client, cluster_arn, service_arn):
+    service = describe_service(ecs_client, cluster_arn, service_arn)
+
+    return [_create_deployment_tuple_from_ecs(service, deployment)
+            for deployment in service['deployments']]
+
+
+def _get_date_string(date):
+    return date.strftime(DATE_FORMAT)
```

Comment:
I went back on forth on this, as I think the function name is more descriptive and makes the code clear, it is also used in two places.